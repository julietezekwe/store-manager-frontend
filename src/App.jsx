/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import './App.css';
import { BrowserRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import Routes from './Routes.jsx';
import NavBar from './components/common/NavBar.jsx';
import Footer from './components/common/Footer.jsx';


export class App extends Component {
    handleLogout = () => {
      localStorage.removeItem('user');
      localStorage.removeItem('token');
      window.location.reload();
    }

    render() {
      const { user } = this.props.login;
      return (
            <React.Fragment>
                <NavBar user={user} logout={this.handleLogout} />
                <BrowserRouter>
                <Routes />
                </BrowserRouter>
                <Footer />
              </React.Fragment>

      );
    }
}

const mapStateToProps = state => ({
  login: state.login,
});

export default connect(mapStateToProps)(App);
