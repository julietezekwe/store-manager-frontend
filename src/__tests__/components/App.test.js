/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import App from '../../App.jsx';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<App />);
});
describe('App Component', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
