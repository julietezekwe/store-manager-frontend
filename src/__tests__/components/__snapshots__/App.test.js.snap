// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should match snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(App)
    login={
      Object {
        "isloggedIn": true,
        "user": Object {
          "email": "admin@gmail.com",
          "id": 1,
          "name": "admin",
          "role": "admin",
          "username": "admin",
        },
      }
    }
  >
    <App
      dispatch={[Function]}
      login={
        Object {
          "isloggedIn": true,
          "user": Object {
            "email": "admin@gmail.com",
            "id": 1,
            "name": "admin",
            "role": "admin",
            "username": "admin",
          },
        }
      }
    >
      <NavBar
        logout={[Function]}
        user={
          Object {
            "email": "admin@gmail.com",
            "id": 1,
            "name": "admin",
            "role": "admin",
            "username": "admin",
          }
        }
      >
        <header
          id="header"
        >
          <nav>
             
            <div
              className="logo"
            >
              <i
                className="fab fa-stripe-s"
              />
              tore 
              <i
                className="fab fa-medium-m"
              />
              anager
            </div>
            <ul>
              <i />
              <li>
                <i
                  className="fas fa-user-lock"
                />
                admin
              </li>
              <li>
                <i
                  className="fas fa-sign-in-alt"
                >
                   
                  <span
                    onClick={[Function]}
                  >
                     Logout
                  </span>
                </i>
              </li>
            </ul>
          </nav>
        </header>
      </NavBar>
      <BrowserRouter>
        <Router
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <Routes>
            <Switch>
              <Route
                component={[Function]}
                computedMatch={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                default={true}
                exact={true}
                location={
                  Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  }
                }
                path="/"
              >
                <Connect(Login)
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "pathname": "/",
                      "search": "",
                      "state": undefined,
                    }
                  }
                  match={
                    Object {
                      "isExact": true,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                >
                  <Login
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "createHref": [Function],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "pathname": "/",
                          "search": "",
                          "state": undefined,
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    location={
                      Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      }
                    }
                    login={
                      Object {
                        "isloggedIn": true,
                        "user": Object {
                          "email": "admin@gmail.com",
                          "id": 1,
                          "name": "admin",
                          "role": "admin",
                          "username": "admin",
                        },
                      }
                    }
                    loginUser={[Function]}
                    match={
                      Object {
                        "isExact": true,
                        "params": Object {},
                        "path": "/",
                        "url": "/",
                      }
                    }
                  >
                    <section
                      className="login-page"
                    >
                      <div
                        className="login-form"
                      >
                        <h1
                          className="login-heading"
                        >
                          Log In
                        </h1>
                        <div
                          className="myCard col-9"
                          id="alert"
                        />
                        <div
                          className="form login"
                        >
                          <div>
                            <input
                              id="email"
                              onChange={[Function]}
                              placeholder="Email"
                              required={true}
                              type="email"
                            />
                          </div>
                          <div>
                            <input
                              id="password"
                              onChange={[Function]}
                              placeholder="password"
                              required={true}
                              type="password"
                            />
                          </div>
                          <div>
                            <input
                              onClick={[Function]}
                              type="submit"
                              value="LOG IN"
                            />
                          </div>
                          <div
                            className="forgotpassword"
                          >
                            <p>
                              <i>
                                Forgot Password?
                              </i>
                            </p>
                          </div>
                        </div>
                      </div>
                    </section>
                  </Login>
                </Connect(Login)>
              </Route>
            </Switch>
          </Routes>
        </Router>
      </BrowserRouter>
      <Footer>
        <footer
          className="col-12"
          id="footer"
        >
          <p>
            Copyright Â© 2018 Store managers Design. All rights reserved.
          </p>
        </footer>
      </Footer>
    </App>
  </Connect(App)>
</Provider>
`;
